TOOLS_OVERVIEW.md

```diff
@@ -0,0 +1,254 @@
+## Overview of CProver Tools
+
+This document provides a brief overview of the various tools in
+the CProver project. The goal is as to provide a lightweight reference
+with links to detailed documentation on each of the tools.
+
+The tools in the CProver project are as follows:
+- [cbmc](#cbmc)
+- [goto-analyzer](#goto-analyzer)
+- [goto-cc (goto-gcc, goto-ld)](#goto-cc)
+- [goto-diff](#goto-diff)
+- [goto-harness](#goto-harness)
+- [goto-instrument](#goto-instrument)
+- [janalyzer](#janalyzer)
+- [jbmc](#jbmc)
+- [jdiff](#jdiff)
+- [memory-analyzer](#memory-analyzer)
+- [smt2_solver](#smt2_solver)
+- [symtab2gb](#symtab2gb)
+- [Developer Utilities](#developer-utilities)
+    - [java-unit](#java-unit)
+    - [unit](#unit)
+    - [others (converter, driver, file_converter, java-converter)](#others)
+
+The rest of this document provides a section on each of these tools in alphabetical order.
+Most links in this document are to the [CProver online documentation](http://cprover.diffblue.com/index.html).
+
+
+## cbmc
+
+The C Bounded Model Checker (`cbmc`) is the main tool used in the CProver suite.
+`cbmc` does the entire analysis from the source code through to the result,
+including generating traces. This includes invoking various sub-tools and 
+modules.
+
+For details on usage of the `cbmc` tool see the following documentation
+- [Developer Tutorial](http://cprover.diffblue.com/tutorial.html)
+includes a very brief tutorial on many aspects of `cbmc` and other tools.
+
+For details on the architecture of the `cbmc` tool and how the analysis is performed
+see the following documents:
+- [CBMC Architecture](http://cprover.diffblue.com/cbmc-architecture.html)
+gives a high level overview of the `cbmc` architecture and data flow.
+- [Background Concepts](http://cprover.diffblue.com/background-concepts.html)
+overviews all the key concepts used in the `cbmc` analysis.
+
+For details on compiling, testing, contributing, and documentation related to 
+development see:
+- [CProver Developer Documentation](http://cprover.diffblue.com/index.html)
+
+
+## goto-analyzer
+
+This provides a way to access and invoke various forms of analysis on
+C source files.
+
+Documentation useful for this tool can be found:
+- [Analysis Information](http://cprover.diffblue.com/group__analyses.html)
+
+Details of all the options can be seen by running
+```
+goto-analyzer --help
+```
+
+## goto-cc
+
+This is a compiler designed to be able to be dropped in to an existing
+build process to replace the existing compiler. `goto-cc` is able to do
+normal compilation, but is designed to output goto programs (optionally
+in addition to the normal compiled program). Note that `goto-cc` is the
+compiler and linker while `goto-gcc` and `goto-ld` are symbolic links to
+`goto-cc`. The additional object code is used as the internal representation
```

Comment:
I'm not sure the fact that these are symlinks matters the most to a user. Perhaps focus on the resulting behaviour? `goto-cc` behaves differently, depending on the file name: the command-line options (and ensuing behaviour) are modelled after the program that has the same name as the suffix, i.e., `gcc`, `ld`, etc. (we also support `cl`, `link`, `bcc`, `as`).

Reply:
This has been updated now to clarify the role of the different files and de-emphasise the symlinks.